### PR TITLE
Upgrade to ReactJS 0.13.x

### PR DESCRIPTION
**Pull request by philwills**
Don't worry, you can use ReactJS without JSX. However, we do reserve the right to make the syntax completely horrible if you don't.

cc @robertberry just in case he has a cunning plan to avoid this

### CODE REVIEW

**Comment by robertberry** [comment]
Not sure what's going on here. Why are you dropping the caps?

**Comment by robertberry** [comment]
Also CoffeeScript is so last month. Get on ES6!

File: app/views/main.scala.html [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
         <script src='@routes.Assets.at("lib/jquery/jquery.min.js")' type="text/javascript"></script>
 
-        <script src='@routes.Assets.at("lib/react/react-with-addons.min.js")'></script>
+        <script src='@routes.Assets.at("lib/react/react-with-addons.js")'></script>
```
Comment (theefer) [inline app/views/main.scala.html]: Why?

Comment (philwills) [inline app/views/main.scala.html]: Because I forgot I'd switched that, as it adds more debugging output.  Thanks for the catch.

**Comment by philwills** [comment]
See https://gist.github.com/sebmarkbage/ae327f2eda03bf165261

Basically, I can't call the classes directly any more, I have to go via something created via `React.createFactory`. This is hidden away by JSX transforms. The cap-dropping was my convention for distinguishing between the two, as I couldn't face having `Factory` littered round the code more than absolutely necessary.

**Comment by philwills** [comment]
I'll get on the ES6 bandwagon when someone writes a transpiler plugin for SBT. I did actually start doing that with Traceur, but it was packaged horribly and so was not fun to wrestle with. Perhaps Babel will be easier.

**Comment by robertberry** [comment]
@philwills You only need to compile the code when building it on TeamCity. It's probably better to have a devmode, that just loads the System.js code and then lets JSPM do its magic inserting script tags on the page itself. That way you don't ever have to run `gulp` or whatever yourself.

**Comment by robertberry** [comment]
@philwills I've not done that yet on the quiz builder but I'm gonna at some point as compilation is slow

**Comment by philwills** [comment]
@robertberry Coffeescript transpiling is really fast

**Comment by theefer** [comment]
There seem to be sbt-traceur plugins, but really, trying to build an asset pipeline is sbt is just masochism (from experience).

**Comment by robertberry** [comment]
@philwills It is with JSPM / Babel if you do it inline in the way I was suggesting. (Rather than compiling the single executable file up front using a build tool.)

**Comment by philwills** [comment]
@theefer for something simple like this, it works really well these days.  I'm not claiming it's the right solution, generally, but for me here it's a good fit. I probably should just bit the bullet and move to ScalaJS though.

**Pull request merged**